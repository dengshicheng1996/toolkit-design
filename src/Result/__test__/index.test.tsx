import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { AlipayCircleFill } from '@ant-design/mobile-icons'
import { Result } from '..'

describe('Button', () => {
  it('snapshot', () => {
    const component = shallow(
      <Result
        img={<AlipayCircleFill color="#1677ff" size="lg" />}
        title="操作成功"
        message="content"
        buttons={[
          {
            type: 'primary',
            children: 'Back Home',
          },
          {
            children: 'Cancel',
          },
        ]}
      />,
    )
    const component2 = shallow(
      <div>
        <Result type="success" message="content" />
        <Result type="error" message="content" />
        <Result type="info" message="content" />
        <Result type="wait" message="content" />
        <Result type="warn" message="content" />
      </div>,
    )

    expect(toJSON(component)).toMatchSnapshot()
    expect(toJSON(component2)).toMatchSnapshot()
  })
})
