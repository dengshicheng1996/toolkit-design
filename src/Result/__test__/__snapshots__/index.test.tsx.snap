// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div
  className="amd-result"
  role="alert"
>
  <div
    className="amd-result-main"
  >
    <div
      className="amd-result-img"
    >
      <AlipayCircleFill
        color="#1677ff"
        size="lg"
      />
    </div>
    <div
      className="amd-result-title"
    >
      操作成功
    </div>
    <div
      className="amd-result-message"
    >
      content
    </div>
  </div>
  <div
    className="amd-result-buttons"
  >
    <Button
      key="0"
      type="primary"
    >
      Back Home
    </Button>
    <Button
      key="1"
    >
      Cancel
    </Button>
  </div>
</div>
`;

exports[`Button snapshot 2`] = `
<div>
  <Result
    message="content"
    type="success"
  />
  <Result
    message="content"
    type="error"
  />
  <Result
    message="content"
    type="info"
  />
  <Result
    message="content"
    type="wait"
  />
  <Result
    message="content"
    type="warn"
  />
</div>
`;
